### ISSUE TITLE
Investigate Path Details memory usage and performance

### ISSUE THREAD

**Issue by karussell** [graphhopper/graphhopper#1162]
Currently there is a measurable slow down of nearly 10% when instructions are enabled. (In the Measurement class we also enable the PathDetails calculation if withInstructions=true).

So we should look if everything is good when looking into the memory and speed via a profiler. Also see if the slow down disappears if path details are disabled.

**Comment by karussell** [comment]
I'm pretty sure that there is no slow down if we disable path details (we still have to check), but I would like to be sure that when we enable them the overhead is as low as possible.

**Comment by boldtrn** [comment]
Ah OK, so this is only when we enable PathDetails? 10% for CH sounds actually not too unreasonable to me, when using a couple of PathDetails. The simplification becomes more complicated, which I guess should be a big part of the slow down. The more PathDetails are requested, the more complicated the simplification becomes. I think I would have expected 5% slow down, just as a gut feeling. For example the edgeId PathDetails, should add a lot more simplification runs.

**Comment by karussell** [comment]
Hmmh, even if I disable path details the same slow down happens, this should not be. Can you confirm this?

**Comment by boldtrn** [comment]
I compared the current master to a state before the PathDetails and indeed the routingCH.mean is about 10% higher in the current master.

I did some profiling and found that the method `PointList#copy` takes too much CPU time. The method is called frequently in `PathSimplification#reduceLength`. This call is removed in #1138, [here](https://github.com/graphhopper/graphhopper/pull/1138/files#diff-0948f2a2464f0901289e39074f54c493L158).

Also the method DouglasPeucker#subSimplify and DouglasPeucker#compressNew take up a lot of cpu time. When comparing it to the older state of GraphHopper, the simplify algorithm took a lot of cpu time before, so this is probably OK.

So I think we could improve the performance having a more efficient `PointList#copy` method. I will try this.
